### PR TITLE
Creates fakeScheduledInterviewDao that is used for testing servlet

### PR DESCRIPTION
**Pull request by GrantComm**

### CODE REVIEW

File: src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import java.util.Set;
+
+/** Mimics accessing Datastore to support managing ScheduledInterview entities. */
+public class FakeScheduledInterviewDao implements ScheduledInterviewDao {
+  // data is the fake database
+  public Map<String, ScheduledInterview> data;
+
+  /** Initializes the fields for ScheduledInterviewDatastoreDAO. */
+  public FakeScheduledInterviewDao() {
+    data = new LinkedHashMap<String, ScheduledInterview>();
```
Comment (meagancheese) [inline src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java]: I think the keys should be Longs.

File: src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+          || email.equals(scheduledInterview.intervieweeEmail())) {
+        relevantInterviews.add(scheduledInterview);
+      }
+    }
+    return relevantInterviews;
+  }
+
+  /**
+   * Returns a list of all scheduledInterviews ranging from minTime to maxTime of a user. minTime
+   * and maxTime are in milliseconds.
```
Comment (meagancheese) [inline src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java]: Remove the minTime and maxTime are in milliseconds comment.

**Review by meagancheese — CHANGES_REQUESTED** [review]
One major change and then a minor change.

File: src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java (hunk 1 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java]: Done

File: src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java (hunk 2 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java]: Done

**Review by meagancheese — APPROVED** [review]
LGTM

File: src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        scheduledInterviewsInRange.add(scheduledInterview);
+      }
+    }
+    return scheduledInterviewsInRange;
+  }
+
+  /** Creates a ScheduledInterview Entity and stores it. */
+  @Override
+  public void create(ScheduledInterview scheduledInterview) {
+    long generatedId = new Random().nextLong();
```
Comment (apluscs) [inline src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java:107]: `generatedId` doesn't this give the possibility of hash collisions?

**Review by apluscs — APPROVED** [review]
LGTM.

**Review by acmcarther — APPROVED** [review]

**Pull request merged**